### PR TITLE
fix: make lib mode `:standard` match defaults

### PR DESCRIPTION
**Pull request by jchavarri**
It seems the `:standard` mode for libraries was not including the "inherited" values for `byte` mode.

This would lead to some errors like the following, when trying to build libraries for an executable with explicit `byte` mode:

```
File "_none_", line 1:
Error: Module `Mylib' is unavailable (required by `Dune__exe__Main')
```

The redundancy of default values makes the code a bit brittle, I added comments to prevent future breakages.

### CODE REVIEW

File: src/dune_rules/dune_file.ml
Comment (anmonteiro) [inline src/dune_rules/dune_file.ml:464]: I'd add a `Mode_conf.Set.Default.to_list` or something, and we'd be synced by default.

**Review by anmonteiro — APPROVED** [review]
LGTM pending my suggestion. Looks like a bug I introduced in https://github.com/ocaml/dune/pull/6611

**Comment by rgrinberg** [comment]
Before merging, would it be possible to split the PR into two commits:

1. First commit that introduces the test and demonstrates that it fails.
2. A second commit that fixes the bug and modifies the test introduced in 1

File: src/dune_rules/dune_file.ml
Comment (jchavarri) [reply src/dune_rules/dune_file.ml]: Fixed in https://github.com/ocaml/dune/pull/7467/commits/b21c4a3710b86a8458a0e88c25cc4c455a467245.

**Comment by jchavarri** [comment]
> would it be possible to split the PR into two commits

Done in https://github.com/ocaml/dune/pull/7467/commits/83387d183641b3336f0d085bd39007c46f4f7cc1 and https://github.com/ocaml/dune/pull/7467/commits/b21c4a3710b86a8458a0e88c25cc4c455a467245.

**Pull request merged**
